### DISCUSSION TITLE
Opening a Grasshopper file containing home-made clusters

### DISCUSSION THREAD

**Discussion by xPhilippeRoux** [HALRobotics/Beta#89] [Ideas]
Hello,

I have a Grasshopper project containing clusters in which various HAL components are located. Having the possibility since Rhino 6 to open the Grasshopper file by double-clicking on it, I'm getting used to that and each time it brings a series of unrecognized objects :

![image](https://user-images.githubusercontent.com/80822085/112624845-6c353c00-8e2e-11eb-8309-1ff8b9dda650.png)

It is yet not the case when you open Rhino first, then Grasshopper, and open your project, where the framework stays functional.

Just so that you know ;)

Philippe

**Comment by sebandraos** [comment]
Hi @XtreeE-PRO ,
This is a known issue that, unfortunately, we don't currently have any means to fix. 
To speed up Grasshopper loading times we load some of our components in the background. This usually only takes half a second or so (depending on computer speed) but that's long enough for Grasshopper to have tried to load the file and not found the components we haven't yet loaded. As you have noted, this doesn't happen when you open Grasshopper manually because the loading is normally faster than your reactions to open a file from the Grasshopper window.

It is also worth noting that very few of our components are cluster-compatible (see #48 )

**Reply by xPhilippeRoux** [reply]
I had actually tested a year ago and clusters were not possible at all. However, it seems to work well now with a large amount of HAL components encapsulated (if of course I open the software first).

**Reply by sebandraos** [reply]
Good to know, thanks @XtreeE-PRO
